src/construction/postcode_construction.py

```diff
@@ -44,6 +44,8 @@ def postcode_data_construction(
     updated_snapshot_df = snapshot_df.copy()
 
     # Add flags to indicate row was constructed
+    if "is_constructed" not in updated_snapshot_df.columns:
```

Comment:
The flag "is_constructed" has been created earlier, in [here](https://vscode.dev/github/ONSdigital/research-and-development/blob/RDRP-1017_pcode_construction_mapping/src/construction/all_data_construction.py#L83-L84). Why/when did it disappear? And why recreating it here helps in mapping the itls? Would we overwrite construction flag for non-postcode construction?

Reply:
If 'all data construction' has been run earlier in the pipeline, then the 'is_constructed' flag will be present in snapshot when it gets to postcode construction.  The issue is when it is only postcode construction being run - in this case, the is_constructed flag isn't in the snapshot, so isn't being updated for the records that undergo postcode construction (and so they are not being picked up by the mask in itl mapping).  I've added this line so if it 'all data construction' isn't being run, the is_constructed column is added, allowing the appropriate flag is given to those records.